autoload/ale/handlers/hdl_checker.vim

```diff
@@ -0,0 +1,70 @@
+" Author:      suoto <andre820@gmail.com>
+" Description: Adds support for HDL Code Checker, which wraps vcom/vlog, ghdl
+"              or xvhdl. More info on https://github.com/suoto/hdl_checker
+
+call ale#Set('hdl_checker_executable', 'hdl_checker')
+call ale#Set('hdl_checker_config_file', has('unix') ? '.hdl_checker.config' : '_hdl_checker.config')
+call ale#Set('hdl_checker_options', '')
+
+" Use this as a function so we can mock it on testing. Need to do this because
+" test files are inside /testplugin (which refers to the ale repo), which will
+" always have a .git folder
+function! ale#handlers#hdl_checker#IsDotGit(path) abort
+    return isdirectory(a:path) && ! empty(a:path)
+endfunction
+
+" Sould return (in order of preference)
+" 1. Nearest config file
+" 2. Nearest .git directory
+" 3. The current path
+function! ale#handlers#hdl_checker#GetProjectRoot(buffer) abort
+    let l:project_root = ale#path#FindNearestFile(
+    \   a:buffer,
+    \   ale#Var(a:buffer, 'hdl_checker_config_file'))
+
+    if !empty(l:project_root    )
+        return fnamemodify(l:project_root, ':h')
+    endif
+
+    " Search for .git to use as root
+    let l:project_root = ale#path#FindNearestDirectory(a:buffer, '.git')
+
+    if ale#handlers#hdl_checker#IsDotGit(l:project_root)
+        return fnamemodify(l:project_root, ':h:h')
+    endif
+
+    " As a fallback, use the path of the current buffer
+    return expand('#' . a:buffer . ':p:h')
+endfunction
+
+function! ale#handlers#hdl_checker#GetCommand(buffer) abort
+    let l:command = ale#Var(a:buffer, 'hdl_checker_executable') . ' --lsp'
```

Comment:
I just noticed, you should apply `ale#Escape()` to the executable path here, in case the path contains spaces, etc.